packages/base-driver/lib/basedriver/commands/execute.js

```diff
@@ -0,0 +1,58 @@
+// @ts-check
+import _ from 'lodash';
+import {makeArgs, checkParams} from '../../protocol';
+
+/**
+ * @param {SessionBase} Base
+ * @returns {ExecuteBase}
+ */
+export function ExecuteMixin(Base) {
+  /**
+   * @implements {IExecuteCommands}
+   */
+  class ExecuteCommands extends Base {
+    /**
+     * @this {Driver}
+     * @param {string} script
+     * @param {Array<Record<string, any>>} protoArgs
+     */
+    async executeOverload(script, protoArgs) {
+      // the w3c protocol will give us an array of arguments to apply to a javascript function.
+      // that's not what we're doing. we're going to look for a JS object as the first arg, so we
+      // can perform validation on it. we'll ignore everything else.
+      let args = protoArgs[0];
+      if (!args) {
+        args = {};
+      }
+      if (!_.isPlainObject(args)) {
+        throw new Error(`Did not receive an appropriate executeOverload parameters object`);
+      }
+
+      const Driver = /** @type {DriverClass} */ (this.constructor);
+      const availableScripts = _.keys(Driver.executeOverloadMap);
+      if (!availableScripts.includes(script)) {
+        throw new Error(
+          `Unsupported overloaded execute script '${script}'. Available scripts ` +
+            `are: ${JSON.stringify(availableScripts)}`
+        );
+      }
+      let argsToApply = [];
+      const commandMetadata = Driver.executeOverloadMap[script];
+      if (!commandMetadata.params) {
+        commandMetadata.params = {required: [], optional: []};
+      }
+      checkParams(commandMetadata.params, args, null);
```

Comment:
if `!commandMetadata.params` is `true`, what is the use of calling `checkParams` the "empty" one we created?

Reply:
there is no use